File: src/PsiCollaboratorManager/Scripts/Lib/Collaborator/CollaboratorFull.js
class CollaboratorFull {
    constructor(
        collaboratorId,
        firstName,
        lastName,
        operatorNumber,
        email,
        dniCollaborator,
        dateOfBirth,
        gender,
        parent,
        maritalStatus,
        telephone1,
        telephone2,
        curriculumFile,
        rfidCode,
        needPasswordChange,
        isLockedOut,
        lockOutEndTime,
        password,
        createUserAccount,
        isActive,
        districtId,
        cantonId,
        provinceId,
        addressLine,
        bankId,
        currencyTypeId,
        numberBankAccount,
        ibanAccount,
        diseases,
        takingMedications,
        note,
        picture,
        emergencyContacts
    ) {
        this.CollaboratorId = collaboratorId;
        this.FirstName = firstName;
        this.LastName = lastName;
        this.OperatorNumber = operatorNumber;
        this.Email = email;
        this.DNICollaborator = dniCollaborator;
        this.DateOfBirth = dateOfBirth;
        this.Gender = gender;
        this.Parent = parent;
        this.MaritalStatus = maritalStatus;
        this.Telephone1 = telephone1;
        this.Telephone2 = telephone2;
        this.CurriculumFile = curriculumFile;
        this.RFIDCode = rfidCode;
        this.NeedPasswordChange = needPasswordChange;
        this.IsLockedOut = isLockedOut;
        this.LockOutEndTime = lockOutEndTime;
        this.Password = password;
        this.CreateUserAccount = createUserAccount;
        this.IsActive = isActive;
        this.DistrictId = districtId;
        this.CantonId = cantonId;
        this.ProvinceId = provinceId;
        this.AddressLine = addressLine;
        this.BankId = bankId;
        this.CurrencyTypeId = currencyTypeId;
        this.NumberBankAccount = numberBankAccount;
        this.IBANAccount = ibanAccount;
        this.Diseases = diseases;
        this.TakingMedications = takingMedications;
        this.Note = note;
        this.Picture = picture;
        this.EmergencyContacts = emergencyContacts;
    }
    static FromGui(emergencyContacts, base64Image) {
        const getValueById = (id, type = 'string') => {
            const element = document.getElementById(id);
            if (!element) return null;

            switch (type) {
                case 'int':
                    return parseInt(element.value, 10) || 0;
                case 'bool':
                    return element.checked;
                case 'date':
                    console.log(element.value);
                    console.log(new Date(element.value.split('-').join('/')));
                    return element.value ? new Date(element.value.split('-').join('/')) : null;
                default:
                    return element.value || '';
            }
        };

        return new CollaboratorFull(
            getValueById('CollaboratorId', 'int'),
            getValueById('FirstName'),
            getValueById('LastName'),
            getValueById('OperatorNumber', 'int'),
            getValueById('Email'),
            getValueById('DNICollaborator'),
            getValueById('DateOfBirth', 'date'),
            document.getElementById('rbtnFemale').checked ? 1 : 2,
            getValueById('rbtnYesMarital', 'bool'),
            getValueById('rbtnYesParent', 'bool'),
            getValueById('Telephone1'),
            getValueById('Telephone2'),
            "",
            getValueById('RFIDCode'),
            false,
            false,
            new Date(),
            getValueById('Password'),
            getValueById('CreateUserAccount', 'bool'),
            getValueById('IsActive', 'bool'),
            getValueById('DistrictId', 'int'),
            getValueById('CantonId', 'int'),
            getValueById('ProvinceId', 'int'),
            getValueById('AddressLine'),
            getValueById('BankId', 'int'),
            getValueById('CurrencyTypeId', 'int'),
            getValueById('NumberBankAccount'),
            getValueById('IBANAccount'),
            getValueById('Diseases'),
            getValueById('TakingMedications', 'bool'),
            getValueById('Note'),
            base64Image,
            emergencyContacts
        );
    }
}

File: src/PsiCollaborator.Data/Collaborator/CollaboratorFull.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PsiCollaborator.Data.Collaborator
{
    public class CollaboratorFull : ICollaboratorFull
    {
        public int CollaboratorId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int OperatorNumber { get; set; }
        public string Email { get; set; }
        public string DNICollaborator { get; set; }
        public DateTime DateOfBirth { get; set; }
        public int Gender { get; set; }
        public bool Parent { get; set; }
        public int? MaritalStatusId { get; set; }
        public string Telephone1 { get; set; }
        public string Telephone2 { get; set; }
        public string CurriculumFile { get; set; }
        public string RFIDCode { get; set; }
        public bool NeedPasswordChange { get; set; }
        public bool IsLockedOut { get; set; }
        public DateTime LockOutEndTime { get; set; }
        public string Password { get; set; }
        public bool CreateUserAccount { get; set; }
        public bool IsActive { get; set; }
        public int DistrictId { get; set; }
        public int CantonId { get; set; }
        public int ProvinceId { get; set; }
        public string AddressLine { get; set; }
        public int BankId { get; set; }
        public int CurrencyTypeId { get; set; }
        public string NumberBankAccount { get; set; }
        public string IBANAccount { get; set; }
        public string Diseases { get; set; }
        public bool TakingMedications { get; set; }
        public string Note { get; set; }
        public string Picture { get; set; }
        public string[] EmergencyContacts { get; set; }
        //public DateTime PasswordChangedDate { get; set; }
        public void SetDefaultValues()
        {
            CollaboratorId = CollaboratorId != 0 ? CollaboratorId : 0;
            OperatorNumber = OperatorNumber != 0 ? OperatorNumber : 0;
            FirstName = FirstName ?? "";
            LastName = LastName ?? "";
            Email = Email ?? "";
            DNICollaborator = DNICollaborator ?? "";
            DateOfBirth = DateOfBirth != DateTime.MinValue ? DateOfBirth : DateTime.MinValue;
            Gender = Gender != 0 ? Gender : 1;
            Parent = Parent;
            MaritalStatusId = MaritalStatusId ?? 1;
            Telephone1 = Telephone1 ?? "";
            Telephone2 = Telephone2 ?? "";
            CurriculumFile = CurriculumFile ?? "";
            RFIDCode = RFIDCode ?? "";
            NeedPasswordChange = NeedPasswordChange;
            IsLockedOut = IsLockedOut;
            LockOutEndTime = LockOutEndTime != DateTime.MinValue ? LockOutEndTime : DateTime.MinValue;
            IsActive = IsActive;
            DistrictId = DistrictId != 0 ? DistrictId : 1;
            CantonId = CantonId != 0 ? CantonId : 101;
            ProvinceId = ProvinceId != 0 ? ProvinceId : 10101;
            AddressLine = AddressLine ?? "";
            BankId = BankId != 0 ? BankId : 1;
            CurrencyTypeId = CurrencyTypeId != 0 ? CurrencyTypeId : 1;
            NumberBankAccount = NumberBankAccount ?? "";
            IBANAccount = IBANAccount ?? "";
            Diseases = Diseases ?? "";
            TakingMedications = TakingMedications;
            Note = Note ?? "";
            Picture = Picture ?? "";
            EmergencyContacts = EmergencyContacts ?? new string[0];
        }
    }
}
